### PR TITLE
fixes #7766 - MS DHCP code scales poorly

### PR DESCRIPTION
**Pull request by pikelly**
Looking at the history of this ticket it is un clear whether another solution is required but, anyway, here is some code that we use in our organization to address this problem. This has been used in a production since 2013 but obviously I have forward ported it so it needs further testing in its new plugin based container running on 1.9 ruby.

### CODE REVIEW

**Comment by dmitri-d** [comment]
I'd prefer to wait with these changes -- I'm getting close to being done with a rewrite of ms dhcp module to use native ms dhcp interface (via ffi). If that proves too slow or problematic for w/e reason, we can come back to this?

**Comment by pikelly** [comment]
An ffi based fast solution would be absolutely great! A caching solution may be required on top of this because even using the M$ snapin things can be a little slow :-( I am look forward to your commit.

File: modules/dhcp_native_ms/dhcp_native_ms_main.rb [hunk 1]
```diff
@@ -195,11 +198,12 @@ def load_subnets
       service.add_subnets(*find_all_subnets)
     end
 
-    def execute cmd, msg=nil, error_only=false
-      tsecs = 5
+    def execute cmd, msg=nil, error_only=false, dumping=false
+      tsecs = 10
```
Comment (alexjfisher) [inline modules/dhcp_native_ms/dhcp_native_ms_main.rb:202]: I've also found that 5 seconds isn't enough...

**Comment by dmitri-d** [comment]
If you are curious: got to the point when ruby bindings are functional and cover most of the functionality required: https://github.com/witlessbird/ruby-dhcpsapi. Pls. see integration tests for examples for now.

**Comment by dmitri-d** [comment]
closing in favour of https://github.com/theforeman/smart-proxy/pull/445

**Pull request closed without merging**